cpp/src/parquet/stream_reader.h

```diff
@@ -140,17 +195,60 @@ class PARQUET_EXPORT StreamReader {
   int64_t SkipRows(int64_t num_rows_to_skip);
 
  protected:
+  using node_ptr_type = std::shared_ptr<schema::PrimitiveNode>;
+
+  [[noreturn]] void ThrowReadFailedException(const node_ptr_type& node);
+
   template <typename ReaderType, typename T>
   void Read(T* v) {
     const auto& node = nodes_[column_index_];
     auto reader = static_cast<ReaderType*>(column_readers_[column_index_++].get());
-
+    int16_t def_level;
+    int16_t rep_level;
     int64_t values_read;
 
-    reader->ReadBatch(1, NULLPTR, NULLPTR, v, &values_read);
+    reader->ReadBatch(1, &def_level, &rep_level, v, &values_read);
```

Comment:
why this change?  it seems like they still aren't used?  did you mean to consolidate code below?

Reply:
The behaviour of ReadBatch() is not the same for optional columns.  And yes I know this is not the function which is not normally called for optional fields.

Let's say that the schema is changed so that a field can be optional.   Existing applications continue to read the field as a required field.   This can work so long as the field is set _and_ we do not use NULLPTR.

Reply:
Thanks for the explanation, I think it is worth adding a comment to this effect.

Reply:
Ok